ports/geckolib/properties.mako.rs

```diff
@@ -666,6 +659,23 @@ fn static_assert() {
         }
     }
 
+    fn set_vertical_align(&mut self, v: longhands::vertical_align::computed_value::T) {
+        <% keyword = data.longhands_by_name["vertical-align"].keyword %>
+        use style::properties::longhands::vertical_align::computed_value::T;
+        // FIXME: Align binary representations and ditch |match| for cast + static_asserts
+        match v {
+            % for value in keyword.values_for('gecko'):
+                T::${to_rust_ident(value)} =>
+                    self.gecko.mVerticalAlign.set_int(structs::${keyword.gecko_constant(value)} as i32),
+            % endfor
+            T::LengthOrPercentage(v) => self.gecko.mVerticalAlign.set(v),
+        }
+    }
+    fn copy_vertical_align_from(&mut self, other: &Self) {
+        self.gecko.mVerticalAlign.mUnit = other.gecko.mVerticalAlign.mUnit;
```

Comment:
it would be good to debug_assert against eStyleUnit_Calc (which is reference-counted and thus the only nsCoord type that cannot be copied by value).

There are a few other checks for the same thing in the other copy_ routines. Maybe abstract out a helper called debug_assert_unit_safe_to_copy(unit: nsStyleUnit) or something?